src/ui/dispatcher.coffee

```diff
@@ -421,6 +421,9 @@ handle 'handlesyncedevents', (r) ->
             conv.addChatMessage e
     connection.setEventState connection.IN_SYNC
 
+handle 'syncrecentconversationsnow', ->
```

Comment:
what's the reason for creating this handler?

Reply:
For what I remember when the conversation goes on/off the record we get a conversation event from hangouts that signals the change, but not a actual message.

This only happens after re-syncing the last messages. I didn't want to create an exception when handling conversation_events so a trigger to the `syncrecentconversations` is performed, but it  has a 10second timeout before running

so `syncrecentconversationsnow` was born 🔥

I didn't try setting the previous handler with a default argument `..., (timeout = 10000) -> ...` so that it preserves compatibility

which could be a better solution, what do you think?

Reply:
That makes sense. I just didn't see any calls to syncrecentconversationsnow action, so I'm wondering if there's any use for it if we're not calling it? Or maybe the part of the code that calls this was not committed.

Reply:
oh!! great eyes!!

Corrected it now.

Reply:
I believe the call to syncrecentconversationsnow is causing an issue that shows up at least half of the time when sending a message. When you send a message, it will normally display the text  as light gray (while it's sending), and then actually display it as normal black text once it's sent. The issue that shows up here is that a lot of the times the text will show light gray, then disappear completely, and then show up as black text. The amount of time it disappears varies, from very short amount of time to very noticeable.

Reply:
Yup, I'm experiencing the same. I've started to work on something a little more intelligent, i.e. update the specific conversation instead of resyncing everything.

Also, the call should only be performed on specific situations and not always _(as it it now)_

Can you change the review to `request changes` ? so that it is shown as not ready?